### PR TITLE
FrodoKEM: Test Specification

### PR DESCRIPTION
**Pull request by reneme**
It contains what it says on the box...

### CODE REVIEW

File: docs/testspec/src/12_pubkey_kem.rst [hunk 1]
```diff
@@ -1,6 +1,107 @@
...
+demanded by the NIST submission and provided by the reference implementation.
+
+Additionally, Botan has implementation-specific test cases. Those ensure the
+interoperability of the algorithm when using Botan's generic API for public key
+algorithms. These test cases are equal for all public key schemes and are
+therefore not discussed in detail in this chapter.
+
+All FrodoKEM-specific test code can be found in
+:srcref:`src/tests/test_frodokem.cpp`. Relevant test data vectors for the KAT
+tests are in *src/tests/data/pubkey/frodokem_kat.vec.vec*.
```
Comment (atreiber94) [inline docs/testspec/src/12_pubkey_kem.rst:17]:
```suggestion
tests are in *src/tests/data/pubkey/frodokem_kat.vec.
```

File: docs/testspec/src/12_pubkey_kem.rst [hunk 2]
```diff
@@ -1,6 +1,107 @@
...
+.. table::
+   :class: longtable
+   :widths: 20 80
+
+   +------------------------+-------------------------------------------------------------------------+
+   | **Test Case No.:**     | PKENC-FRODO-2                                                           |
+   +========================+=========================================================================+
+   | **Type:**              | Positive/Negative Test                                                  |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Description:**       | For each implemented (and available) mode: generate a randome key pair  |
```
Comment (atreiber94) [inline docs/testspec/src/12_pubkey_kem.rst:75]:
```suggestion
   | **Description:**       | For each implemented (and available) mode: generate a random key pair  and |
```

**Review by atreiber94 — COMMENTED** [review]
Looks good! I only found typos and have the following remarks:

- Do we need to specify the Keygen test?
- Do we need to specify what is tested depending on the modules?

**Comment by reneme** [comment]
I didn't include the keygen test, because, in my mind, the described KAT and roundtrip tests should provide enough coverage. Regarding build configuration: I would aim to explain what happens if we build with default configuration (i.e. the BSI build policy). Nevertheless, that's a valid point, thanks.

**Review by atreiber94 — APPROVED** [review]
lgtm

**Review by sehlen-bsi — APPROVED** [review]

**Comment by reneme** [comment]
@sehlen-bsi thanks for the review. 😄

I'm going to merge this into `main`, as FrodoKEM will be part of the next release. LMS PRs will have to stay open until it is clear which release of Botan they will land in.

**Pull request merged**